plugins/nexus-repository-conda/src/main/java/org/sonatype/nexus/repository/conda/internal/hosted/CondaHostedFacetImpl.java

```diff
@@ -0,0 +1,262 @@
+import org.sonatype.nexus.repository.conda.internal.hosted.metadata.PackageDesc;
+import org.sonatype.nexus.repository.conda.internal.hosted.metadata.RepoData;
+import org.sonatype.nexus.repository.conda.internal.util.CondaDataAccess;
+import org.sonatype.nexus.repository.config.Configuration;
+import org.sonatype.nexus.repository.storage.*;
+import org.sonatype.nexus.repository.transaction.TransactionalDeleteBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalStoreBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalStoreMetadata;
+import org.sonatype.nexus.repository.transaction.TransactionalTouchBlob;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.ContentTypes;
+import org.sonatype.nexus.repository.view.Payload;
+import org.sonatype.nexus.repository.view.payloads.StringPayload;
+import org.sonatype.nexus.transaction.Transactional;
+import org.sonatype.nexus.transaction.UnitOfWork;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.StreamSupport;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.Collections.singletonList;
+import static org.sonatype.nexus.repository.conda.internal.hosted.Utils.toAttributes;
+import static org.sonatype.nexus.repository.conda.internal.hosted.metadata.MetaData.asIndex;
+import static org.sonatype.nexus.repository.conda.internal.hosted.metadata.MetaData.readIndexJson;
+import static org.sonatype.nexus.repository.conda.internal.util.CondaDataAccess.HASH_ALGORITHMS;
+import static org.sonatype.nexus.repository.conda.internal.util.CondaDataAccess.toContent;
+import static org.sonatype.nexus.repository.storage.ComponentEntityAdapter.P_GROUP;
+import static org.sonatype.nexus.repository.storage.ComponentEntityAdapter.P_VERSION;
+import static org.sonatype.nexus.repository.storage.MetadataNodeEntityAdapter.P_NAME;
+
+/**
+ * {@link CondaHostedFacetImpl implementation}
+ */
+@Named
+public class CondaHostedFacetImpl
+        extends FacetSupport
+        implements CondaHostedFacet {
+    private final CondaDataAccess condaDataAccess;
+
+    @Inject
+    public CondaHostedFacetImpl(final CondaDataAccess condaDataAccess) {
+        this.condaDataAccess = checkNotNull(condaDataAccess);
+    }
+
+    @Override
+    protected void doInit(Configuration configuration) throws Exception {
+        super.doInit(configuration);
+        getRepository().facet(StorageFacet.class).registerWritePolicySelector(new CondaWritePolicySelector());
+    }
+
+    @Override
+    @TransactionalTouchBlob
+    public Optional<Content> fetch(final String path) {
+        checkNotNull(path);
+
+        StorageTx tx = UnitOfWork.currentTx();
+        Asset maybeAsset = condaDataAccess.findAsset(tx, tx.findBucket(getRepository()), path);
+
+        return Optional.ofNullable(maybeAsset)
```

Comment:
If you are returning a possible `null` you should likely add `@Nullable` from `javax.annotation` to the method for IDEs, etc...

Reply:
The whole point of Optional is to eliminate the possibility of return null.
Type system is looking after the correctness of handling missing asset situations here.